tests/expectations/metrics/test_core.py

```diff
@@ -76,8 +77,18 @@ def test_basic_metric_pd():
     assert results == {desired_metric.id: 3}
 
 
-def test_mean_metric_pd():
-    engine = build_pandas_engine(pd.DataFrame({"a": [1, 2, 3, None]}))
+@pytest.mark.parametrize(
+    "dataframe,result",
+    [
+        [pd.DataFrame({"a": [1, 2, 3, None]}), 2],
+        [
+            pd.DataFrame({"a": [Decimal(2.0), Decimal(0.18781)]}),
+            1.093905,
+        ],
+    ],
+)
```

Comment:
@TrangPham This is a nice parametrization.  How would you feel if we also tried to include a case where a column contains both, integers and decimals in different rows (right now, all types are the same) -- and see if it passes?  Thanks.

Reply:
@TrangPham On a second thought, columns with values that have different types (heterogeneous) are rare/unlikely.  So if this does not look meaningful after an initial quick experiments, let us abandon this and stick to homogeneous (same-type valued) columns (like you have it already).  Thank you.